libs/features/personalization/promo-utils.js

```diff
@@ -1,27 +1,51 @@
-import { getMetadata } from '../../utils/utils.js';
+import { getMetadata, getConfig } from '../../utils/utils.js';
 
 const GMTStringToLocalDate = (gmtString) => new Date(`${gmtString}+00:00`);
 
+const APAC = ['au', 'cn', 'hk_en', 'hk_zh', 'id_en', 'id_id', 'in', 'in_hi', 'kr', 'my_en', 'my_ms', 'nz', 'ph_en', 'ph_fil', 'sg', 'th_en', 'th_th', 'tw', 'vn_en', 'vn_vi'];
+const EMEA = ['ae_en', 'ae_ar', 'africa', 'at', 'be_en', 'be_fr', 'be_nl', 'bg', 'ch_de', 'ch_fr', 'ch_it', 'cis_en', 'cis_ru', 'cz', 'de', 'dk', 'ee', 'eg_ar', 'eg_en', 'es', 'fi', 'fr', 'gr_el', 'gr_en', 'hu', 'ie', 'il_en', 'il_he', 'iq', 'is', 'it', 'kw_ar', 'kw_en', 'lt', 'lu_de', 'lu_en', 'lu_fr', 'lv', 'mena_ar', 'mena_en', 'ng', 'nl', 'no', 'pl', 'pt', 'qa_ar', 'qa_en', 'ro', 'ru', 'sa_en', 'sa_ar', 'se', 'si', 'sk', 'tr', 'ua', 'uk', 'za'];
+const AMERICAS = ['us', 'ar', 'br', 'ca', 'ca_fr', 'cl', 'co', 'cr', 'ec', 'gt', 'la', 'mx', 'pe', 'pr'];
+const JP = ['jp'];
+const REGIONS = { APAC, EMEA, AMERICAS, JP };
+
+const getLocaleCode = () => {
+  const getLocale = () => getConfig()?.locale?.prefix?.substring(1) || 'us';
+  return (getLocale()?.split('-')?.pop() || 'us').toLowerCase();
+};
+
+const localeCode = getLocaleCode();
+const regionCode = Object.keys(REGIONS)
+  .find((r) => REGIONS[r]?.includes(localeCode))?.toLowerCase() || null;
+
 export const isDisabled = (event, searchParams) => {
   if (!event) return false;
+  if (event.locales && !event.locales.includes(localeCode)) return true;
   const currentDate = searchParams?.get('instant') ? new Date(searchParams.get('instant')) : new Date();
   if ((!event.start && event.end) || (!event.end && event.start)) return true;
   return Boolean(event.start && event.end
     && (currentDate < event.start || currentDate > event.end));
 };
 
-export default function getPromoManifests(manifestNames, searchParams) {
+const isManifestWithinLocale = (locales) => {
+  if (locales) {
+    return locales.split(';').map((locale) => locale.trim()).includes(localeCode);
+  }
+  return true;
```

Comment:
this means if the excel locales cell is empty if will apply to all geos, right?
this is probably ok but needs to be documented.
same as the fact that the only can use
/**/... once

Reply:
yes, typically we won't ask locales in "default" manifest.
/**/ once per region!